crates/ruff_linter/src/rules/pyupgrade/rules/unnecessary_encode_utf8.rs

```diff
@@ -256,3 +263,78 @@ pub(crate) fn unnecessary_encode_utf8(checker: &Checker, call: &ast::ExprCall) {
         _ => {}
     }
 }
+
+fn string_contains_string_only_escapes(string: &StringLiteralValue, locator: &Locator) -> bool {
+    for fragment in string {
+        let flags = fragment.flags;
+
+        if flags.prefix().is_raw() {
+            continue;
+        }
+
+        let total_len = fragment.end() - fragment.start();
+        let value_len = total_len - flags.opener_len() - flags.closer_len();
+
+        if value_len > fragment.as_str().text_len() {
+            return string_fragment_contains_string_only_escapes(fragment, locator);
+        }
+    }
+
+    false
+}
+
+fn string_fragment_contains_string_only_escapes(
+    fragment: &StringLiteral,
+    locator: &Locator,
+) -> bool {
+    let flags = fragment.flags;
+
+    let inner_start = fragment.start() + flags.opener_len();
+    let inner_end = fragment.end() - flags.closer_len();
+    let inner_in_source = locator.slice(TextRange::new(inner_start, inner_end));
```

Comment:
Let's add a `content_range` method to `StringLiteral`. It seems generally useful, similar to 

https://github.com/astral-sh/ruff/blob/ef85c682bdd76ed63457bdfbec72e09424ac20ab/crates/ruff_python_ast/src/expression.rs#L221-L228